recommender/recommender.py

```diff
@@ -443,36 +474,35 @@ def add_resource(self, data, _suffix=''):
         for field in self.resource_content_fields:
             result[field] = data[field]
 
+        resource_id = stem_url(data['url'])
         # check url for redundancy
-        for recommendation in self.recommendations:
-            if self.check_redundancy(recommendation['url'], data['url']):
-                result['error'] = ('The resource you are attempting to ' +
-                                   'provide has already existed')
-                for field in self.resource_content_fields:
-                    result['dup_' + field] = recommendation[field]
-                result['dup_id'] = recommendation['id']
-                result['Success'] = False
-                tracker.emit('add_resource', result)
-                return result
-            
+        if resource_id in self.recommendations:
+            result['error'] = ('The resource you are attempting to ' +
+                               'provide has already existed')
+            for field in self.resource_content_fields:
+                result['dup_' + field] = self.recommendations[resource_id][field]
+            result['dup_id'] = self.recommendations[resource_id]['id']
+            result['Success'] = False
```

Comment:
You should not use `result['Success'] = False` to indicate a failed HTTP request. You should use the [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) to indicate the error state, and ideally provide a human-readable message as well explaining what went wrong.